app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/network/MarvelRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.network
+
+import com.naxtlevelofandroiddevelopment.marvelgallery.model.MarvelCharacter
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.Provider
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.RetrofitApi
+import io.reactivex.Single
+
+interface MarvelRepository {
+
+    fun getCharacters(searchQuery: String?): Single<List<MarvelCharacter>>
```

Comment:
Yes, not about method naming but more about not bloating responsibilities for getting data with optional criteria